### PR TITLE
Add: Kerberos credential for targets

### PR DESCRIPTION
**Pull request by timopollmeier**
## What
Targets now have the krb5_credential element that can be used to pass Kerberos 5 credentials to the scanner for authenticated scans.

## Why
This allows scans of Windows systems that no longer allow the old SMB authentication method.

## References
GEA-775

### CODE REVIEW

File: src/gmp.c [hunk 1]
```diff
@@ -23134,6 +23202,31 @@ gmp_xml_handle_end_element (/* unused */ GMarkupParseContext* context,
...
+              if (send_find_error_to_client
+                   ("create_target", "Credential",
+                    create_target_data->snmp_credential_id,
+                    gmp_parser))
+                {
+                  error_send_to_client (error);
+                  return;
+                }
+            }
+          else if (create_target_data->smb_credential_id
```
Comment (mattmundell) [inline src/gmp.c:23223]: Minor: This check could be before the SMB check on 23131, to skip the SMB and KRB processing when both are given.

File: src/manage_sql.c [hunk 2]
```diff
@@ -32937,6 +32940,22 @@ create_target (const char* name, const char* asset_hosts_filter,
            new_target, snmp_credential, "0");
     }
 
+  if (krb5_credential)
+    {
+      gchar *type = credential_type (krb5_credential);
+      if (strcmp (type, "krb5"))
+        {
+          sql_rollback ();
```
Comment (mattmundell) [inline src/manage_sql.c:32948]: This should free `type`. The existing cases have the same, so could be a new PR.

File: src/manage_sql.c [hunk 3]
```diff
@@ -33544,6 +33570,54 @@ modify_target (const char *target_id, const char *name, const char *hosts,
...
+          if (krb5_credential == 0)
+            {
+              sql_rollback ();
+              return 26;
+            }
+
+          type = credential_type (krb5_credential);
+          if (strcmp (type, "krb5"))
+            {
+              sql_rollback ();
```
Comment (mattmundell) [inline src/manage_sql.c:33602]: Same here with the free of `type`.

File: src/schema_formats/XML/GMP.xml.in [hunk 4]
```diff
@@ -22074,6 +22091,37 @@ END:VCALENDAR
...
+            <e>trash</e>
+          </pattern>
+          <ele>
+            <name>name</name>
+            <summary>The name of the Kerberos 5 LSC credential</summary>
+            <pattern><t>name</t></pattern>
+          </ele>
+          <ele>
+            <name>permissions</name>
+            <summary>Permissions the user has on the task</summary>
```
Comment (mattmundell) [inline src/schema_formats/XML/GMP.xml.in:22116]:
```suggestion
            <summary>Permissions the user has on the credential</summary>
```

**Review by mattmundell — CHANGES_REQUESTED** [review]
See single comments above. Happy to do the low priority ones in separate PR.

File: src/gmp.c [hunk 5]
```diff
@@ -23134,6 +23202,31 @@ gmp_xml_handle_end_element (/* unused */ GMarkupParseContext* context,
...
+                       (create_target_data->krb5_credential_id,
+                        &krb5_credential,
+                        "get_credentials"))
+            SEND_TO_CLIENT_OR_FAIL (XML_INTERNAL_ERROR ("create_target"));
+          else if (create_target_data->krb5_credential_id
+                   && krb5_credential == 0)
+            {
+              if (send_find_error_to_client
+                   ("create_target", "Credential",
+                    create_target_data->snmp_credential_id,
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/gmp.c:23216]: Fixed in [0e49e1e](https://github.com/greenbone/gvmd/pull/2332/commits/0e49e1ec1be168cb7aa179f28150dc0a93541d52)

File: src/gmp.c (hunk 1 above)
Comment (timopollmeier) [reply src/gmp.c:23223]: Fixed in [0e49e1e](https://github.com/greenbone/gvmd/pull/2332/commits/0e49e1ec1be168cb7aa179f28150dc0a93541d52)

File: src/manage_sql.c (hunk 2 above)
Comment (timopollmeier) [reply src/manage_sql.c:32948]: Fixed in [0e49e1e](https://github.com/greenbone/gvmd/pull/2332/commits/0e49e1ec1be168cb7aa179f28150dc0a93541d52)

File: src/manage_sql.c (hunk 3 above)
Comment (timopollmeier) [reply src/manage_sql.c:33602]: Fixed in [0e49e1e](https://github.com/greenbone/gvmd/pull/2332/commits/0e49e1ec1be168cb7aa179f28150dc0a93541d52)

File: src/schema_formats/XML/GMP.xml.in (hunk 4 above)
Comment (timopollmeier) [reply src/schema_formats/XML/GMP.xml.in]: I've fixed this and several similar instances incorrectly referring to a task in [0f82c2e](https://github.com/greenbone/gvmd/pull/2332/commits/0f82c2e93b6495ee404acd0ac65ca85e8b9371e8)

**Review by mattmundell — APPROVED** [review]

**Pull request merged**